### PR TITLE
docs(middleware.md): translated & proofread 'middleware.md'

### PR DESCRIPTION
**Pull request by lslxdx**
'middleware.md' is translated by @lslxdx and proofread by @闷油瓶小张

related issue: #783 

##### Checklist
- [x] documentation is changed or added

##### Description of change

close https://github.com/eggjs/egg/pull/783

### CODE REVIEW

File: docs/source/en/basics/middleware.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
+title: Middleware
+---
+
+In [the previous chapter](../intro/egg-and-koa.md), we say that Egg is based on Koa 1, so the form of middleware in Egg is the same as in Koa 1, i.e. they are both based on [the onion model](../intro/egg-and-koa.md#midlleware) of generator function.
+
+## Writing Middleware
+
+### how to write
```
Comment (popomore) [inline docs/source/en/basics/middleware.md]: capitalize `how`

File: docs/source/en/basics/middleware.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+We begin by writing a simple gzip middleware, to see how to write middleware
```
Comment (popomore) [inline docs/source/en/basics/middleware.md]: We begin to see how to write a middleware from a simple gzip example

File: docs/source/en/basics/middleware.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  this.body = stream;
+  this.set('Content-Encoding', 'gzip');
+}
+```
+
+You might find that the middleware's writing style in the framework is exactly the same as in Koa, so any middleware in Koa can be used directly by the framework.
+
+### Configuration
+
+Usually the middleware has its own configuration. In the framework, a complete middleware is including the configuration process. We agree that a middleware is a separate file placed in `app/middleware` directory, which needs an exports function that take two parameters:
```
Comment (popomore) [inline docs/source/en/basics/middleware.md]: a middleware is a file in `app/middleware` directory by convention

File: docs/source/en/basics/middleware.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    this.body = stream;
+    this.set('Content-Encoding', 'gzip');
+  };
+};
+```
+
+## Importing Middleware in the Application
+
+We can import customized middleware completely by configuration in the application, and decide their order.
+If we need to import the gzip Middleware in the above,
```
Comment (popomore) [inline docs/source/en/basics/middleware.md]: Middleware > middleware

File: docs/source/en/basics/middleware.md [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+In addition to the application layer middleware is imported, the framework itself and other plug-ins will also import many middleware. All the config fields of these built-in middlewares can be modified by modifying the ones with the same name in the config file, for example [Framework Built-in Plugin](https://github.com/eggjs/egg/tree/master/app/middleware) uses a bodyParser middleware(the framework loader will change the file name separated by delimiters into the camel style), and we can add configs below in `config/config.default.js` to modify the bodyParser:
+
+```js
+module.exports = {
+  bodyParser: {
+    jsonLimit: '10m',
+  },
+};
+```
+** Note: middleware imported by the framework and plugins are loaded earlier than those imported by the application layer, and the application layer cannot overwrite the default framework middleware. If the application layer imports customized middleware that has the same name with default framework middleware, an error will be raised on starting up. **
```
Comment (popomore) [inline docs/source/en/basics/middleware.md]: better changing `layer` to `middleware`

File: docs/source/en/basics/middleware.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+## The Use of Middleware in Router
```
Comment (popomore) [inline docs/source/en/basics/middleware.md]:
```diff
- The Use of Middleware in Router
+ Middleware in Router
```

**Comment by popomore** [comment]
感觉中文文档可以把引入改为加载，翻译成 import 有点难懂，还是叫 load 比较好。

File: docs/source/en/basics/middleware.md (hunk 5 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: Change 'application layer' to 'application middleware'?

How about changing 'application layer' to 'application'? Does it make sense?

File: docs/source/en/basics/middleware.md (hunk 1 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: Done. commit: fefcbdbe

File: docs/source/en/basics/middleware.md (hunk 2 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: Done. commit: fefcbdbe

File: docs/source/en/basics/middleware.md (hunk 3 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: Done. commit: fefcbdbe

File: docs/source/en/basics/middleware.md (hunk 4 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: Done. commit: fefcbdbe

File: docs/source/en/basics/middleware.md (hunk 6 above)
Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: Done. commit: fefcbdbe

**Comment by lslxdx** [comment]
@popomore 
> 感觉中文文档可以把引入改为加载，翻译成 import 有点难懂，还是叫 load 比较好。

需要在英文文档中修改过来吗？还是说先改中文文档，再改英文文档？

**Comment by popomore** [comment]
> 需要在英文文档中修改过来吗？还是说先改中文文档，再改英文文档？

中英文一起改吧，看看其他人的意见 @dead-horse @atian25

File: docs/source/en/basics/middleware.md (hunk 5 above)
Comment (popomore) [inline docs/source/en/basics/middleware.md]: LGTM

Comment (lslxdx) [inline docs/source/en/basics/middleware.md]: Good!
Anyway, it is not finalized and can be changed at any time though.

Does anyone have other suggestions?

**Comment by atian25** [comment]
都行吧，小的点顺手改也可以，主要还是 focus 翻译上。其他的我们另开 PR 都行。

发自我的 iPhone

**Comment by lslxdx** [comment]
@popomore @atian25 

可以做3件事情：

1. 创建一个单独的issue, 比如`中文文档应该把引入改为加载，因为翻译成 import 有点难懂，还是叫 load 比较好。`
2. 在这个doc的翻译版上改成"load"
3. 在这个doc的中文版上改成"加载"

可行吗？哪几件(些)事情可行的？

**Comment by popomore** [comment]
@lslxdx 在这篇文章里面把中英文都改了好了，其他文章遇到翻译不通顺的地方可以选择性修改。

**Comment by lslxdx** [comment]
@popomore 好的，晚上改

**Comment by lslxdx** [comment]
@popomore IIRC, we discussed the "application layer/middleware" issue days ago(I can't find the discussion URL, sigh), luckily I find a similar expression [here](http://expressjs.com/en/guide/using-middleware.html) just now, for example: 

> You can load application-level and router-level middleware with an optional mount path. 

and 

> Application-level middleware

So could we use "application-level" as an agreement?

**Comment by popomore** [comment]
When we talk about middleware, we have two steps

1. First middelwares will be loaded from application, plugin and framework, it can be called application-level middleware. build-in middleware also meam framework-level middleware.
1. Then I use the middleware which have been loaded, it can be used automatically by configuration whitch is `config.appMiddleware` and `config.coreMiddleware`, it also can be used in router manually. That's what express does.

So I agree using `xx-level` in document of load middleware, but don't use it in use middleware.

**Comment by lslxdx** [comment]
> So I agree using `xx-level` in document of load middleware, but don't use it in use middleware.

Yes, that makes sense and I agree too!

**Comment by atian25** [comment]
Is there anything still block this pr, guys?

**Comment by popomore** [comment]
@lslxdx can you complete this PR first?

**Comment by lslxdx** [comment]
@popomore I'm stuck on this thread and could you please help me out?

All I can see is:
### Review required
At least one approved review is required by reviewers with write access. Learn more.
### All checks have passed
5 successful checks
### Merging is blocked
Merging can be performed automatically with one approved review.

What should I do next?

**Comment by popomore** [comment]
> > So I agree using xx-level in document of load middleware, but don't use it in use middleware.
>
> Yes, that makes sense and I agree too!

I think you will update something, maybe I was wrong?

**Comment by lslxdx** [comment]
Aha! Got it and I am on it!

**Comment by lslxdx** [comment]
@popomore I make some changes just now and please have a look.

**Pull request merged**